pkg/userdata/amzn2/provider.go

```diff
@@ -140,6 +140,20 @@ package_upgrade: true
 package_reboot_if_required: true
 {{- end }}
 
+{{- if .ProviderSpec.CAPublicKey }}
+- path: "/etc/ssh/sshd_config"
+  content: |
+	TrustedUserCAKeys /etc/ssh/trusted-user-ca-keys.pem
+	CASignatureAlgorithms ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521,ssh-ed25519,rsa-sha2-512,rsa-sha2-256,ssh-rsa
+  append: true
+{{- end }}
+
+{{- if .ProviderSpec.CAPublicKey }}
+- path: "/etc/ssh/trusted-user-ca-keys.pem"
+  content: |
+{{ .ProviderSpec.CAPublicKey | indent 4 }}
+{{- end }}
+
```

Comment:
```suggestion
{{- with .ProviderSpec.CAPublicKey }}
- path: "/etc/ssh/sshd_config"
  content: |
	{{ SshdConfigAddendum }} << this should be defined in FuncMap
  append: true

- path: "/etc/ssh/trusted-user-ca-keys.pem"
  content: |
{{ . | indent 4 }}
{{- end }}
```

Reply:
Done

Reply:
I meant add new function to the https://github.com/kubermatic/machine-controller/blob/b51933b55d3474a6f26014e56bee985a78aa0f13/pkg/userdata/helper/template_functions.go#L27-L46

Reply:
I've done that now

I reverted the change and unresolved this conversation